File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/chain_of_responsibility/lab/CustomerServiceManager.java
// CustomerServiceManager.java

public class CustomerServiceManager
{
	public boolean canHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		return (aCustomerRequestsDiscountEvent.getPercent() < 30);
	}

	public void doHandle(CustomerRequestsDiscountEvent aCustomerRequestsDiscountEvent)
	{
		System.out.println("I am a CustomerServiceManager.");
		System.out.println("I can give you a " + aCustomerRequestsDiscountEvent.getPercent() + "% discount.");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/chain_of_responsibility/lab/CustomerServiceManager.m
% CustomerServiceManager.m

classdef CustomerServiceManager < handle
	methods
		% constructor:
		function theCustomerServiceManager = CustomerServiceManager
		end

		function result = canHandle(theCustomerServiceManager, aCustomerRequestsDiscountEvent)
			result = (aCustomerRequestsDiscountEvent.Percent < 30);
		end

		function doHandle(theCustomerServiceManager, aCustomerRequestsDiscountEvent)
			fprintf('I am a CustomerServiceManager.\n');
			fprintf('I can give you a %d%% discount.\n', aCustomerRequestsDiscountEvent.Percent);
		end
	end
end
